magnet/excitation-data/si-corrector-fc2-ch.txt

```diff
@@ -0,0 +1,78 @@
+# HEADER
+# ======
+# label             si-corrector-fc2-ch
+# harmonics         0 2 8
+# main_harmonic     0 normal
+# rescaling_factor  1.0
+# units             Ampere  T*m T*m  T.m/m^1 T.m/m^1  T.m/m^7 T.m/m^7
+
+# EXCITATION DATA
+# ===============
+-01.0000  -3.1817e-04 -1.0498e-06 +9.8651e-01 -6.7864e-03 +7.1760e+09 -1.6058e+08 
+-00.7500  -2.3873e-04 -7.7331e-07 +7.3977e-01 -4.6804e-03 +5.3151e+09 -1.0511e+08 
+-00.5000  -1.5872e-04 -5.6511e-07 +4.9195e-01 -3.7793e-03 +3.5929e+09 -6.6684e+07 
+-00.2500  -7.9111e-05 -2.3122e-07 +2.4567e-01 -2.5870e-03 +1.7967e+09 -5.0351e+07 
++00.0000  +0.0000e+00 +0.0000e+00 +0.0000e+00 +0.0000e+00 +0.0000e+00 +0.0000e+00 
++00.2500  +7.9111e-05 +2.3122e-07 -2.4567e-01 +2.5870e-03 -1.7967e+09 +5.0351e+07 
++00.5000  +1.5872e-04 +5.6511e-07 -4.9195e-01 +3.7793e-03 -3.5929e+09 +6.6684e+07 
++00.7500  +2.3873e-04 +7.7331e-07 -7.3977e-01 +4.6804e-03 -5.3151e+09 +1.0511e+08 
++01.0000  +3.1817e-04 +1.0498e-06 -9.8651e-01 +6.7864e-03 -7.1760e+09 +1.6058e+08 
```

Comment:
I think these current values are not correct. The excitation curves for FC2 were measured only up to 0.7A and for this value the integrated field is already close to 3e-4 T.m.

Reply:
luana, since we will use averages over corretors for all 4 types i fitted a 2-nd order polynomial and reconstructed the avg excitation curve over a range of currentea which includes measured values.

as fot the integrate field at 0.7A, it is strange. it is the same script (whose link is in the excdata file) that calculate the average for FC1... did you find problem with FC1 as well?

we need to investigate this....

by the way, do people in the IMA group do not already have a script that generates this excdata file? it would have been a nice independent check in the past....

Reply:
FC1 values appear to be correct. But I only compared it with the excitation curve of one random corrector, I didn't average them all.

I think the IMA group doesn't have a script to generate these curves. But I agree with you that these checks would be very useful.

Reply:
@LuanaVilela , only now I had the time to investigate the issue you raised. You were right, there is a bug in how the excdata was being read that affected only FC2 correctors! The bug was a class variable which selects the apropriate 'FC1' | 'FC2' folder in the measurement repository. It was fized for 'FC1', instead of changing o 'FC2' for FC2 excdata analysis. This is why everything was correct for FC1 (CH and CV) and wrong for FC2 (CH and CV). Thanks!